spring-pulsar-test/src/main/java/org/springframework/pulsar/test/support/ConsumedMessagesConditions.java

```diff
@@ -41,42 +42,30 @@ static <T> ConsumedMessagesCondition<T> desiredMessageCount(int messageCount) {
 	}
 
 	/**
-	 * Verifies that any of the consumed messages has a payload that equals the specified
-	 * value.
+	 * Verifies that the expected value equals the message payload value of at least one
+	 * consumed message.
 	 * @param expectation the expected value
 	 * @param <T> the type of the message
 	 * @return the condition
 	 */
-	static <T> ConsumedMessagesCondition<T> anyMessageMatchesExpected(T expectation) {
-		return messages -> messages.stream().anyMatch(message -> message.getValue().equals(expectation));
+	static <T> ConsumedMessagesCondition<T> atLeastOneMessageMatches(T expectation) {
```

Comment:
The naming of these I struggled with until I started using the util. 
> "consume messages until at least one message matches 'a'" 

helps me understand what this is doing. 

Likewise w/ the N variant below, 
> "consume messages until at least one message matches each of 'a', 'b', and 'c'"